### ISSUE TITLE
[BUG] - Playing or shuffling playlist limited to 100 songs

### ISSUE THREAD

**Issue by skarrok** [CappielloAntonio/tempo#258]
## Describe the bug
Playing or shuffling playlist limited to 100 songs

## Expected behavior
Entire playlist shoud be added or shuffled into play queue

## To Reproduce

1. Open playlist with more than 100 songs
2. Press Play or Shuffle
3. Open queue and see only 100 songs

## Additional context
Restriction introduced in this commit https://github.com/CappielloAntonio/tempo/commit/623a4956a5a3765f151db540ed64925211748a1c

Discussion: https://github.com/CappielloAntonio/tempo/discussions/242

**Comment by Wirezat** [comment]
the change seems to be very much intended and needed, so i wouldnt completely revert it. maybe the number should be un hardcoded though and moved to a setting

**Comment by QuantumCoded** [comment]
@Wirezat I agree the change seems intended, but I feel simply limiting the size of a play queue is a non-solution. The inability to queue and shuffle through all of my favorites (~1k songs) is one of the major things holding me back from switching from Ultrasonic to Tempo.

**Comment by LucaSinUnaS** [comment]
As mentioned in the issue, the problem is that the limit is hardcoded to 100 songs. I forked the lastest release and removed the limit. As also mentioned, when clicking to view the play queue UI it lags at first slightly, but with my fork and apk release the limit is removed.

[Here it is for those interested](https://github.com/LucaSinUnaS/tempo/releases/tag/main)

**Comment by shrapnelnet** [comment]
any chance that you'll be contributing back to this repo from your fork? very interested seeing this change go to upstream. 

is threading not an option for loading large playlists without disrupting the UI?

**Comment by shrapnelnet** [comment]
have done some local testing 

inner_fragment_player_queue.xml is potentially problematic: done some messing around, something about nestedscrollview playing rough with recyclerview. will keep testing

when the queue is first opened, API requests are fired for all album covers which I think is the bottleneck here, working on image lazy loading now

**Comment by LucaSinUnaS** [comment]
> any chance that you'll be contributing back to this repo from your fork? very interested seeing this change go to upstream.
> 
> is threading not an option for loading large playlists without disrupting the UI?

Just saw your pr, and added your changes to the fork I made and made a new apk release with your changes! Thank you for continuing with this, it works much better now.
